src/api/server.rkt

```diff
@@ -79,56 +79,98 @@
   (rename-file-or-directory tmp-file data-file #t)
   (copy-file (web-resource "report.html") html-file #t))
 
+(define single-threaded-cache (make-hash))
+
 ; computes the path used for server URLs
 (define (make-path id)
   (format "~a.~a" id *herbie-commit*))
 
 ; Returns #f is now job exsist for the given job-id
 (define (get-results-for job-id)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'result job-id b))
-  (log "Getting result for job: ~a.\n" job-id)
-  (place-channel-get a))
+  (cond
+    [manager
+     (define-values (a b) (place-channel))
+     (place-channel-put manager (list 'result job-id b))
+     (log "Getting result for job: ~a.\n" job-id)
+     (place-channel-get a)]
+    [else (hash-ref single-threaded-cache job-id #f)]))
 
 (define (get-timeline-for job-id)
-  (define-values (a b) (place-channel))
-  (place-channel-put manager (list 'timeline job-id b))
-  (log "Getting timeline for job: ~a.\n" job-id)
-  (place-channel-get a))
+  (cond
+    [manager
+     (define-values (a b) (place-channel))
+     (place-channel-put manager (list 'timeline job-id b))
+     (log "Getting timeline for job: ~a.\n" job-id)
+     (place-channel-get a)]
+    [else
+     (define result (hash-ref single-threaded-cache job-id #f))
+     (if (false? result) (reverse (unbox (*timeline*))) result)]))
```

Comment:
`(if result result (reverse ...))` or just `(and result (reverse ...)`

Reply:
Also is this correct? If there is a result it looks like it'll save it.

Also wait why goes into `single-threaded-cache`? In fact, how can we ever call this method while the timeline is running? Why not just always return `#f`?